src/utils/renderHelpers.ts

```diff
@@ -0,0 +1,108 @@
+import { Node } from '../ast/nodes/shared/Node';
+import MagicString from 'magic-string';
+import { RenderOptions } from '../Module';
+
+export function findFirstOccurrenceOutsideComment (code: string, searchString: string) {
+	let codeStart = 0;
+	let commentStart, commentLength, lineBreakPos;
+	while (true) {
+		commentStart = code.indexOf('/');
+		lineBreakPos = (~commentStart ? code.slice(0, commentStart) : code).indexOf(searchString);
+		if (~lineBreakPos || !~commentStart) {
+			break;
+		}
+		code = code.slice(commentStart + 1);
+		codeStart += commentStart + 1;
+		if (code[0] === '*') {
+			commentLength = code.indexOf('*/') + 2;
+		} else if (code[0] === '/') {
+			commentLength = code.indexOf('\n') + 1;
```

Comment:
I admit I did not consider line-breaks can be one of `\n`, `\r` and `\r\n`. Not yet sure how to handle this most efficiently. I guess we can always assume that the type of line-break does not change within a file so we could detect it on a per-module basis, store it as a variable on the module and feed this in as a parameter? Otherwise, we would  probably  need to replace `.indexOf` with the less efficient `.search` and do additional checking. I'll think about it.

Reply:
I don't know enough about editors to know if there could be cases where some lines have `\r` and others not, but flexible code might be better I think.

One option could be to search backwards for an `\r` after finding the `\n`?

Also if you find yourself wanting to use `.search`, it can just be faster to use a while loop with a custom index and `charCodeAt` checks (faster than string comparison). Often code like this can do with a general tracking index across multiple while loops anyway.

Reply:
Interesting. I did some measurements of my own and I can only confirm this partially. The only thing that seems to be universally true is that a `string.charCodeAt(x)` comparison is much faster than a `string[x]` comparison. Beyond that I compared the following code (iterated over a large array of strings that had the character "a" all at the same given position):
```javascript
// 1
result = x.search(/a|b/);

// 2
result = 0;
var code;
while (code = x.charCodeAt(result), code !== 97 && code !== 98 && result < 50) result++;
```

The result was that (2) was faster when the character was very early in the string. They took about the same time when it was at position 14. When it was at position 50, (1) was more than twice as fast.

When I define the regular expression in a setup block and use it as a variable, (1) becomes even faster with parity reached at about position 12. Considering (2) does not even tell you yet if the RegExp has been matched at all and also requires careful handling of the break-off condition (unless you want to start building infinite loops), I would favour (1) unless I am sure I will be checking only very few characters.